### PR TITLE
Add OpenTelemetry tracing with grpc interceptors

### PR DESCRIPTION
**Pull request by atoulme**
#### Type of change
- New feature

#### Description

Adds OpenTelemetry tracing by adding interceptors on all gRPC communications.

#### Related issues
This is tied to https://github.com/hyperledger/fabric-rfcs/blob/main/text/0000-opentelemetry-tracing.md

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -19,7 +19,7 @@ require (
 	github.com/frankban/quicktest v1.11.3 // indirect
 	github.com/fsouza/go-dockerclient v1.7.0
 	github.com/go-kit/kit v0.9.0
-	github.com/golang/protobuf v1.3.3
+	github.com/golang/protobuf v1.5.2
```
Comment (denyeart) [inline go.mod:23]: We have been stuck at protobuf v1.3 and older grpc due to etcd issues. See background in https://jira.hyperledger.org/browse/FAB-18363 and https://github.com/hyperledger/fabric/pull/2185.

@guoger has been looking at etcd update in https://github.com/hyperledger/fabric/pull/2707 and may be able to fill in the details.

Comment (atoulme) [reply go.mod:23]: After trying a few things, it indeed appears performance is worse with the latest and some substantial changes are required to continue. I will wait for @guoger and hopefully #2027 helps.

**Comment by ale-linux** [comment]
Closing owing to the long inactivity - please re-open and rebase if it's still needed

**Comment by atoulme** [comment]
I cannot reopen this PR, I will open a new one.

**Pull request closed without merging**